tests/inst/src/destructive.rs

```diff
@@ -0,0 +1,630 @@
+    // that leads to probably 300 network requests, so we want
+    // a low average delay.
+    let webserver_opts = TestHttpServerOpts {
+        random_delay: Some(time::Duration::from_secs_f64(WEBSERVER_DELAY_SECS)),
+        ..Default::default()
+    };
+    with_webserver_in(&srvrepo, &webserver_opts, move |addr| {
+        let url = format!("http://{}", addr);
+        sh_execute!(
+            "ostree remote delete --if-exists testrepo
+             ostree remote add --set=gpg-verify=false testrepo {url}",
+            url = url
+        )?;
+
+        if firstrun {
+            // Also disable some services (like zincati) because we don't want automatic updates
+            // in our reboots, and it currently fails to start.  The less
+            // we have in each reboot, the faster reboots are.
+            sh_execute!("systemctl disable --now zincati fedora-coreos-pinger")?;
+            // And prepare for updates
+            sh_execute!("rpm-ostree cleanup -pr")?;
+            generate_update(&commit)?;
+            // Directly set the origin, so that we're not dependent on the pending deployment.
+            // FIXME: make this saner
+            sh_execute!(
+                "
+                ostree admin set-origin testrepo {url} {testref}
+                ostree refs --create testrepo:{testref} {commit}
+                ostree refs --create={origref} {commit}
+                ",
+                url = url,
+                origref = ORIGREF,
+                testref = TESTREF,
+                commit = commit
+            )?;
+            // We gather a single "cycle time" at start as a way of gauging how
+            // long an upgrade should take, so we know when to interrupt.  This
+            // obviously has some pitfalls, mainly when there are e.g. other competing
+            // VMs when we start but not after (or vice versa) we can either
+            // interrupt almost always too early, or too late.
+            let start = time::Instant::now();
+            upgrade_and_finalize().context("Firstrun upgrade failed")?;
+            let end = time::Instant::now();
+            let cycle_time = end.duration_since(start);
+            let tdata = TransactionalTestInfo {
+                cycle_time: cycle_time,
+            };
+            let mut f = std::io::BufWriter::new(std::fs::File::create(&TDATAPATH)?);
+            serde_json::to_writer(&mut f, &tdata)?;
+            f.flush()?;
+            sh_execute!("rpm-ostree status")?;
+        }
+
+        let tdata = {
+            let mut f = std::io::BufReader::new(std::fs::File::open(&TDATAPATH)?);
+            serde_json::from_reader(&mut f).context("Failed to parse test info JSON")?
+        };
+
+        impl_transaction_test(commit.as_str(), &tdata, mark.as_ref())?;
+
+        Ok(())
+    })?;
+    Ok(())
+}
+
+#[itest(destructive = true)]
+fn other() -> Result<()> {
+    testinit()?;
+    Ok(())
+}
```

Comment:
Is this test leftover from testing or is it meaningful? Otherwise, let's save the VM. :)

Reply:
Thanks, removed :recycle: